src/views/Records/Records.vue

```diff
@@ -142,85 +124,88 @@ export default {
     });
-    this.setStickToTopLocal(false);
+    window.scrollTo(0, 0);
   },
   methods: {
     ...mapActions('records', ['fetchRecords']),
-    ...mapActions({setScrollStatusLocal: 'uiController/setScrollStatus'}),
-    ...mapActions({setStickToTopLocal: 'uiController/setStickToTop'}),
-    onScroll: function (e) {
+    ...mapActions('uiController', ['setScrollStatus', 'setStickToTop']),
+    onScroll: function () {
       let _module = this;
-      _module.offsetTop = e.target.scrollTop;
-      if (_module.offsetTop > 100 && _module.records.length > 1) {
-          _module.setStickToTopLocal(true);
-          _module.$store.dispatch("uiController/setGeneralUIAttributesAction", {
-            bodyOverflowState: true,
-            headerVisibilityState: false,
+      _module.offsetTop = window.top.scrollY;
+      if (_module.offsetTop > 100 && _module.records.length > 1)
+      {
+        _module.setStickToTop(true);
+        _module.$store.dispatch("uiController/setGeneralUIAttributesAction", {
+          headerVisibilityState: false,
         });
       }
-      else {
-          _module.setStickToTopLocal(false);
-          _module.$store.dispatch("uiController/setGeneralUIAttributesAction", {
-            bodyOverflowState: true,
-            drawerVisibilityState: false,
-            headerVisibilityState: true,
-          });
+      else
+      {
+        _module.setStickToTop(false);
+        _module.$store.dispatch("uiController/setGeneralUIAttributesAction", {
+          drawerVisibilityState: false,
+          headerVisibilityState: true,
+        });
       }
-      _module.offsetTop > 500 ? _module.setScrollStatusLocal(true) :
-          _module.setScrollStatusLocal(false);
+      _module.offsetTop > 500 ? _module.setScrollStatus(true) :
+          _module.setScrollStatus(false);
     },
     /**
      * Try to redirect to search of the page that is hit is /standards /databases
      * /policies or /collections
      * */
     tryRedirect: async function () {
-      if (Object.keys(this.recordTypes).includes(this.$route.name)) {
+      if (Object.keys(this.recordTypes).includes(this.$route.name))
+      {
         let fairsharingRegistry = this.recordTypes[this.$route.name];
         let query = this.$route.params;
         if (query && query !== {}) {
           query.fairsharingRegistry = fairsharingRegistry;
-          try {
+          try
+          {
             await this.$router.push({
               name: "search",
               query: query
             });
             return true;
           }
-          catch (e) {
+          catch (e)
+          {
             //
           }
         }
       }
-      await this.getData()
+      await this.getData();
+      window.scrollTo({
+        top: 0,
+        left: 0,
+        behavior: 'smooth'
+      });
     },
     /** This methods get the data from the client.
      * @returns {Promise}
      */
     getData: async function () {
-      window.scrollTo(0, 0);
```

Comment:
I would remove all the scroll to top features and have it triggered here, before any query is being trigger so that we always get the scroll effect before the data is reloaded.

Reply:
Thanks.okey I ll move it before getData function.